modules/openid/cms_openid_email_service_parameters.e

```diff
@@ -0,0 +1,260 @@
+				end
+				s := cfg.text_item ("email")
+				if s /= Void then
+					l_contact_email := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_register")
+				if s /= Void then
+					l_subject_register := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_activate")
+				if s /= Void then
+					l_subject_register := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_password")
+				if s /= Void then
+					l_subject_register := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+				s := cfg.text_item ("subject_oauth")
+				if s /= Void then
+					l_subject_oauth := utf.utf_32_string_to_utf_8_string_8 (s)
+				end
+
+			end
+			if l_contact_email /= Void then
+				if not l_contact_email.has ('<') then
+					l_contact_email := l_site_name + " <" + l_contact_email + ">"
+				end
+				contact_email := l_contact_email
+			else
+				contact_email := admin_email
+			end
+			if l_subject_register /= Void then
+				contact_subject_register := l_subject_register
+			else
+				contact_subject_register := "Thank you for registering with us."
+			end
+
+			if l_subject_activate /= Void then
+				contact_subject_activate := l_subject_activate
+			else
+				contact_subject_activate := "New account activation token."
+			end
+			if l_subject_password /= Void then
+				contact_subject_password := l_subject_password
+			else
+				contact_subject_password := "Password Recovery."
+			end
+			if l_subject_oauth /= Void then
+				contact_subject_oauth := l_subject_oauth
+			else
+				contact_subject_oauth := "Welcome."
+			end
+
+		end
+
+feature	-- Access
+
+	smtp_server: IMMUTABLE_STRING_8
+
+	admin_email: IMMUTABLE_STRING_8
+
+	contact_email: IMMUTABLE_STRING_8
+			-- Contact email.
+
+	contact_subject_register: IMMUTABLE_STRING_8
+	contact_subject_activate: IMMUTABLE_STRING_8
+	contact_subject_password: IMMUTABLE_STRING_8
+	contact_subject_oauth: IMMUTABLE_STRING_8
+
+	account_activation: STRING
+			-- Account activation template email message.
+		do
+			Result := template_string ("account_activation.html", default_template_account_activation)
+		end
+
+	account_re_activation: STRING
+			-- Account re_activation template email message.
+		do
+			Result := template_string ("accunt_re_activation.html", default_template_account_re_activation)
+		end
+
+	account_password: STRING
+			-- Account password template email message.
+		do
+			Result := template_string ("account_new_password.html", default_template_account_new_password)
+		end
+
+	account_welcome: STRING
+			-- Account welcome template email message.
+		do
+			Result := template_string ("account_welcome.html", default_template_account_welcome)
+		end
+
+feature {NONE} -- Implementation: Template		
+
+	template_path (a_name: READABLE_STRING_GENERAL): PATH
+			-- Location of template named `a_name'.
+		do
+			Result := setup.environment.config_path.extended ("modules").extended ("login").extended (a_name)
```

Comment:
Please check my previous change on similar module, and prefer using the CMS_API interface.
i.e avoid using setup.environment  ... it should never be useful

in addition this will not work since "login" is not anymore the good name

prefer to use {CMS_AUTHENTICATION_MODULE}.name

but still .. I am not convinced the templates are needed here ...
I forgot about oauth20 module, if we have the same code .. but I would have same comments.